addons/hr_expense/models/hr_expense.py

```diff
@@ -970,6 +996,12 @@ def _compute_attachment_number(self):
         for sheet in self:
             sheet.attachment_number = sum(sheet.expense_line_ids.mapped('attachment_number'))
 
+    @api.depends('expense_line_ids.attachment_ids')
+    def _compute_main_attachment(self):
+        for sheet in self:
+            if not sheet.message_main_attachment_id or sheet.message_main_attachment_id not in sheet.attachment_ids:
+                sheet.message_main_attachment_id = sheet.attachment_ids[:1]
```

Comment:
in 17.0 would change into something like
```suggestion
            if not sheet.message_main_attachment_id or sheet.message_main_attachment_id not in sheet.attachment_ids:
                sheet.message_main_attachment_id = sheet.expense_line_ids[:1].main_attachment_id
```

Reply:
Thanks for the heads up :)